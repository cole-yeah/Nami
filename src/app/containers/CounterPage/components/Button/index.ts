import styled from 'styled-components/macro';

export const Button = styled.button`
  color: ${p => p.theme.primary};
  text-decoration: none;
  font-size: 14px;
  background-color: #ffffff;
  border-radius: 2px;
  border: 1px solid #666;
`;
