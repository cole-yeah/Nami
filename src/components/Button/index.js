import React, { useEffect } from 'react'

import css from './index.module.scss'

const Index = ({
  icon,
  onClick,
  children,
}) => {
  
  useEffect(() => {

  }, [])

  return (
    <div 
      className={css.btn}
      onClick={onClick}>
      
      <div 
        className={css.icon}
        style={{
          display: icon ? 'block':'none'
        }}
      >{icon || '3'}</div>
      {
        children
      }
    </div>
  )
}

export default Index