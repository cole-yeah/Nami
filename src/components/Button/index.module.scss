.btn {
  display: inline-flex;
  border-radius: 2px;
  padding: 4px 10px;
  background-color: rgb(6, 67, 85);
  color: #ffffff;
}
.icon {
  width: 12px;
  height: 12px;
  margin-right: 4px;
}