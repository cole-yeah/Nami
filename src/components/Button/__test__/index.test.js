import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../index'

test('button test', () => {
  const component = renderer.create(
    <Button>按钮</Button>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})