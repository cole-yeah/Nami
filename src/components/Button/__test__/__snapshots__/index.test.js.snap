// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test 1`] = `
<div
  className="btn"
>
  <div
    className="icon"
    style={
      Object {
        "display": "none",
      }
    }
  >
    3
  </div>
  按钮
</div>
`;
